servicetalk-utils-internal/src/main/java/io/servicetalk/utils/internal/DurationUtils.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.utils.internal;
+
+import java.time.Duration;
+
+import static java.time.Duration.ZERO;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Helper utilities for {@link Duration}.
+ */
+public final class DurationUtils {
```

Comment:
This class will have more utility methods in a follow-up PRs

Reply:
I will move the private `Publisher.toNanos(Duration)` here.